app/src/main/java/com/duckduckgo/app/browser/SpecialUrlDetector.kt

```diff
@@ -31,14 +37,17 @@ interface SpecialUrlDetector {
         class Telephone(val telephoneNumber: String) : UrlType()
         class Email(val emailAddress: String) : UrlType()
         class Sms(val telephoneNumber: String) : UrlType()
-        class IntentType(val url: String, val intent: Intent, val fallbackUrl: String?) : UrlType()
+        class AppLink(val appIntent: Intent? = null, val excludedComponents: List<ComponentName>? = null, val url: String? = null) : UrlType()
+        class NonHttpAppLink(val url: String, val intent: Intent, val fallbackUrl: String?) : UrlType()
         class SearchQuery(val query: String) : UrlType()
         class Unknown(val url: String) : UrlType()
     }
-
 }
 
-class SpecialUrlDetectorImpl : SpecialUrlDetector {
+class SpecialUrlDetectorImpl(
+    private val context: Context,
```

Comment:
would it make sense to pass in the `PackageManager` (or our own wrapper around it) so we avoid passing the context around?